src/buildings/resource/building_resource.gd

```diff
@@ -3,43 +3,16 @@ class_name BuildingResource
 
 @export_group("Display")
 @export var name: String
-@export var type: Building.TYPES
+@export var type: EnumAutoload.BuildingType
```

Comment:
What's the benefit of moving this building type enum out of the Building class?

Reply:
Have a central file for all the Enum make changing them easier, we don't have to go and find them. I also can make sure all of them loaded first before any other autoloader can use them, so no bad dependency.